client-common/src/balance/balance_change.rs

```diff
@@ -1,20 +1,46 @@
 use std::ops::Add;
 
 use failure::ResultExt;
+use rlp::{Decodable, DecoderError, Encodable, Rlp, RlpStream};
 
 use chain_core::init::coin::Coin;
 
 use crate::{ErrorKind, Result};
 
 /// Incoming or Outgoing balance change
-#[derive(Debug)]
+#[derive(Debug, Clone, PartialEq)]
 pub enum BalanceChange {
     /// Represents balance addition
     Incoming(Coin),
     /// Represents balance reduction
     Outgoing(Coin),
 }
 
+impl Encodable for BalanceChange {
```

Comment:
Why to RLP-encode/decode these client-internal types?
`Coin` has deserialize/serialize (as it's used in `InitConfig` which needs it for Tendermint's genesis.json) -- so these client-internal types could auto-derive serde traits and use bincode (like elsewhere in the client code)?

Reply:
TxId perhaps doesn't have it, but we may potentially add it or a client could represent it as a raw `[u8;32]` alias rather than that opaque type

Reply:
similarly address

Reply:
But there may be other reason to have RLP here rather than just that those chain-core types didn't have serde traits

Reply:
There's no particular reason to use RLP here. I used it because internal types don't implement `Serialize` and `Deserialize`, and also, to keep encoding and decoding consistent everywhere.

Reply:
For "consistency", client has been mainly using bincode internally, so I assume that would be preferable.
RLP is meant for "external" binary data, what's broadcasted / stored on TM.

So if the client backend exposes any "binary" data externally, it'll make sense in RLP. If it doesn't expose that binary data (only exposes JSON), it'll be more consistent to store internally in bincode.

Anyway, I'll leave it as it is now and we can come back to it later.